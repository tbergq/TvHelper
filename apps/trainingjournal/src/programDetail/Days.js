// @flow

import * as React from 'react';
import { graphql, createFragmentContainer } from '@tbergq/tvhelper-relay';
import styled from 'styled-components';

import type { Days_week as WeekType } from './__generated__/Days_week.graphql';
import Day from './Day';

const DaysContainer = styled.div({
  display: 'flex',
  justifyContent: 'space-between',
  flexWrap: 'wrap',
});

type Props = {|
  +week: ?WeekType,
|};

export const Days = (props: Props) => {
  const days = props.week?.days?.edges ?? [];
  return (
    <div>
      <DaysContainer>
        {days.map(day => (
          <Day key={day?.node?.id} day={day?.node} />
        ))}
      </DaysContainer>
    </div>
  );
};

export default createFragmentContainer(Days, {
  week: graphql`
    fragment Days_week on Week {
      days {
        edges {
          node {
            id
            ...Day_day
          }
        }
      }
    }
  `,
});
