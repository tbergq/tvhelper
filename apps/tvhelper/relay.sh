#!/bin/bash

cp ../graphql/schema.graphql ./schema.graphql
yarn relay-compiler --language typescript