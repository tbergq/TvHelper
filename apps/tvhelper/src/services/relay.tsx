import { ReactNode } from 'react';
import { Environment, Network, RecordSource, Store, FetchFunction } from 'relay-runtime';
import { RelayEnvironmentProvider } from 'react-relay';
import { RecordMap } from 'relay-runtime/lib/store/RelayStoreTypes';

interface Props {
  readonly children: ReactNode;
  readonly records?: RecordMap;
  readonly token: string | undefined;
}

interface CreateEnvironmentArgs {
  fetchQuery: FetchFunction;
  records?: RecordMap;
}
export const createEnvironment = ({ fetchQuery, records }: CreateEnvironmentArgs) => {
  const network = Network.create(fetchQuery);

  const store = new Store(new RecordSource(records));

  return new Environment({
    network,
    store,
  });
};

const { GRAPHQL_URL } = process.env;

const getBody = (operation, variables) => {
  if (operation.id) {
    return {
      queryId: operation.id,
      variables,
    };
  }
  return {
    query: operation.text,
    variables,
  };
};

export const makeFetchQuery = (token, baseUrl = '') => {
  const fetchQuery = async (operation, variables /* , cacheConfig, uploadables */) => {
    const res = await fetch(`${baseUrl}`, {
      method: 'POST',
      headers: {
        // Add authentication and other headers here
        'content-type': 'application/json',
        'authorization': token,
      },
      body: JSON.stringify(getBody(operation, variables)),
    });
    return res.json();
  };
  return fetchQuery;
};

const EnvironmentProvider = ({ children, records, token }: Props) => {
  const environment = createEnvironment({
    fetchQuery: makeFetchQuery(token, GRAPHQL_URL),
    records,
  });

  return <RelayEnvironmentProvider environment={environment}>{children}</RelayEnvironmentProvider>;
};

export { EnvironmentProvider };
