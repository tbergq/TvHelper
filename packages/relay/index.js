// @flow

export { default as QueryRenderer } from './src/QueryRenderer';
export { graphql, createFragmentContainer } from '@kiwicom/relay';
