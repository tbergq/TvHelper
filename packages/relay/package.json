{
  "name": "@tbergq/relay",
  "private": true,
  "main": "index.js",
  "version": "0.0.1",
  "sideEffects": false,
  "dependencies": {
    "@adeira/fetch": "^1.0.2",
    "@adeira/js": "^1.2.1",
    "@adeira/relay": "^1.3.0",
    "@adeira/relay-utils": "^0.7.0",
    "@tbergq/components": "*",
    "graphql": "^15.0.0",
    "react-fast-compare": "^3.1.1"
  },
  "peerDependencies": {
    "react": "",
    "react-relay": "*",
    "relay-runtime": ""
  },
  "devDependencies": {
    "relay-test-utils": "^9.1.0"
  }
}
