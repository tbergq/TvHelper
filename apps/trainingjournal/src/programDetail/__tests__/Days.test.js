// @flow

import * as React from 'react';
import { QueryRenderer, graphql } from '@kiwicom/relay';
import { createMockEnvironment, MockPayloadGenerator } from 'relay-test-utils';
import { create } from 'react-test-renderer';

import Days from '../Days';

const renderInner = props => <Days week={props.program.weeks.edges[0].node} />;

describe('Days', () => {
  it('renders', () => {
    const environment = createMockEnvironment();
    const TestRenderer = () => (
      <QueryRenderer
        environment={environment}
        query={graphql`
          query DaysQuery @relay_test_operation {
            program(programId: "123") {
              weeks {
                edges {
                  node {
                    ...Days_week
                  }
                }
              }
            }
          }
        `}
        variables={{}}
        onResponse={renderInner}
      />
    );
    const renderer = create(<TestRenderer />);
    environment.mock.resolveMostRecentOperation(operation =>
      MockPayloadGenerator.generate(operation),
    );

    expect(renderer).toMatchSnapshot();
  });
});
