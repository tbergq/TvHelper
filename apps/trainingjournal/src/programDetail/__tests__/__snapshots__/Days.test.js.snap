// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Days 1`] = `
<div>
  <div
    className="sc-cSHVUG kqukHW"
  >
    <div
      className="sc-jzJRlG TAPDy"
    >
      <h3
        className="Heading__StyledHeading-sc-1b8cso5-0 kyjUks"
      >
        <a
          href="/programs/day?dayId=<Day-mock-id-1>"
          onClick={[Function]}
        >
          &lt;mock-value-for-field-"name"&gt;
        </a>
      </h3>
      <div>
        &lt;mock-value-for-field-"name"&gt;
      </div>
    </div>
  </div>
</div>
`;
